### ISSUE TITLE
Support constraintsFiltering=strict  for maven

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#25961]
### Describe the proposed change(s).

Ref: #25718

Renovate should look for the `java.version` property in the repo `pom.xml` and report it as `constraints.java`. Meanwhile the maven datasource should also look for the same and return it as part of the datasource result too. The combination of both of these will allow constraintsFiltering=strict to work for Maven

**Comment by rarkins** [comment]
Reproduction: https://github.com/renovate-reproductions/25961

**Comment by rarkins** [comment]
No, let's stick to the simpler scope I described for this issue. When people insist a solution is perfect or nothing, then most of the time you get nothing. You are welcome to submit a PR for this or any future improvement

**Comment by viral4u** [comment]
> No, let's stick to the simpler scope I described for this issue. When people insist a solution is perfect or nothing, then most of the time you get nothing. You are welcome to submit a PR for this or any future improvement

Can you provide me some pointers on how to go about implementing this? How its done for other datasources like NPM? How to get minimum required runtime version of Java jars?

**Comment by rarkins** [comment]
You need both the maven manager as well as maven datasource to return a constraints.java value. In the npm manager/datasource it's constraints.node

**Comment by ddbtrmatic** [comment]
In the following example, right now renovate only seems to match on the `version` field, and as a result, version `2.0.3-arm64` is chosen, despite `win64` being specified in the dependency.

```xml
    <dependency>
      <groupId>foo</groupId>
      <artifactId>bar</artifactId>
      <version>2.0.1</version>
      <classifier>win64</classifier>
      <type>zip</type>
    </dependency>
```

```
remote/foo/bar/2.0.1/2.0.1-win64.zip
remote/foo/bar/2.0.2/2.0.2-win64.zip
remote/foo/bar/2.0.3/2.0.3-arm64.zip
```

**Comment by Boulea7** [comment]
I'd like to work on the scoped `java.version` behavior described here: propagate the repository POM property into `extractedConstraints.java`, and attach each release POM's property as `constraints.java` when strict filtering is requested. I'll keep other Maven Java-version mechanisms and the unrelated classifier case out of scope.
